cli/util/file_watcher.rs

```diff
@@ -376,28 +396,42 @@ fn new_watcher(
         .iter()
         .filter_map(|path| canonicalize_path(path).ok())
         .collect();
+
       sender.send(paths).unwrap();
     },
     Default::default(),
   )?)
 }
 
-fn add_paths_to_watcher(watcher: &mut RecommendedWatcher, paths: &[PathBuf]) {
+fn add_paths_to_watcher(
+  watcher: &mut RecommendedWatcher,
+  paths: &[PathBuf],
+  paths_to_exclude: &Arc<Mutex<Vec<PathBuf>>>,
+) {
   // Ignore any error e.g. `PathNotFound`
+  let paths_to_exclude = paths_to_exclude.lock().clone();
+  let mut watched_paths = Vec::new();
+
   for path in paths {
+    if paths_to_exclude.iter().any(|p| path.ends_with(p)) {
```

Comment:
Are these paths normalized in any way? Checking just `ends_with()` seems error-prone if paths are not fully resolved.

Reply:
Hi @bartlomieju, thanks for reviewing that!

Good point, they are normalized, I changed it to equality check.